doc/api/process.md

```diff
@@ -416,6 +416,55 @@ generate a core file.
 
 This feature is not available in [`Worker`][] threads.
 
+## process.allowedNodeEnvironmentFlags
+<!-- YAML
+added: REPLACEME
+-->
+
+* {Set}
+
+The `process.allowedNodeEnvironmentFlags` property is a special,
+read-only `Set` of flags allowable within the [`NODE_OPTIONS`][]
+environment variable.
+
+`process.allowedNodeEnvironmentFlags` extends `Set`, but overrides
```

Comment:
I think this is an implementation detail that the person reading the docs doesn't really need to know, I'd just concentrate on describing the behavior of the api, e.g., how to use `.has()` to check for the flag.

Reply:
I think mentioning that it's a Set subclass is important, and is more than just an implementation detail - it's the observable interface.